src/components/calcite-label/calcite-label.tsx

```diff
@@ -30,6 +30,9 @@ export class CalciteLabel {
   //
   //--------------------------------------------------------------------------
 
+  /** specify the for(htmlFor) attribute (optional) */
+  @Prop({ reflect: true }) htmlFor: string;
```

Comment:
Question - does this need to be explicitly defined as a prop?  It should be spread if provided in the DOM to the rendered label via https://github.com/Esri/calcite-components/blob/c46d1264aa17f62fe262d14e045f462026ef7506/src/components/calcite-label/calcite-label.tsx#L191

Reply:
I had to add that to make typescript happy.
```
[ ERROR ]  TypeScript: ./src/components/calcite-radio-button/calcite-radio-button.tsx:243:11
           Type '{ dir: string; htmlFor: string; scale: "s" | "m" | "l"; }' is not assignable to type 'CalciteLabel &
           HTMLAttributes<HTMLCalciteLabelElement>'.Property 'htmlFor' does not exist on type 'CalciteLabel &
           HTMLAttributes<HTMLCalciteLabelElement>'.

    L242:  dir={document.documentElement.getAttribute("dir")}
    L243:  htmlFor={this.guid}
    L244:  scale={this.scale}
```

Reply:
@macandcheese I really don't like the attribute spread approach. Stencil components are prop-based, which may or may not get reflected. Why can't we pass the props onto directly to children?

Reply:
Yeah I'm a fan of explicit prop passing.  I wrote a Select React component a while ago that spread props onto a third-party select element and I found it hard to debug because there were no prop unsupported warnings like Typescript is doing here.  What also ends up happening is props that the `<label>` element doesn't understand or support end up getting passed to it unnecessarily.

Reply:
@jcfranco @eriklharper We definitely can, I think we do in a lot of places. There are just components that could potentially accept dozens of props which we'd need to define in order to have them passed (which, is actually what Ionic itself does in cases, so maybe we should just follow that pattern closer - think each explicitly defined props for any possible input attribute: https://github.com/ionic-team/ionic/blob/master/core/src/components/input/input.tsx)

I guess a use case like `calcite-button` where we render a child element based on the presence of a requested prop is a better example, for instance if a `href` is passed the resultant child element is actually an anchor, not a button, so spreading things like aria-*, title, target, etc. there seemed a little cleaner than accepting those explicitly on the `calcite-button`? 

I think there may have been something around "on___" events that needed to be attached to rendered child elements vs. the wrapping `<Host>` that was solved by just spreading them.

BUT! I do think it's probably safer to just declare it explicitly in this case, we can go back and update some of the components that have spread attributes to explicitly accept them instead.

Reply:
Turns out there's no need to explicitly add this as a prop now.  See comment https://github.com/Esri/calcite-components/pull/482#discussion_r422259668